apps/sens/fsw/unit_test/sens_config_tbl_test.cpp

```diff
@@ -48,15 +49,191 @@
 #include "ut_cfe_fs_stubs.h"
 #include "ut_cfe_time_stubs.h"
 
-void SENS_Config_Tbl_Test_Case1(void)
+
+/**************************************************************************
+ * Tests for SENS InitConfigTbl()
+ **************************************************************************/
+/**
+ * Test SENS InitConfigTbl(), fail TBL Register
+ * Can not call private function, InitConfigTbl directly
+ */
+void Test_SENS_InitConfigTbl_Fail_TblRegister(void)
 {
+    SENS  oSENS;
+
+    int32 result = CFE_SUCCESS;
+    int32 expected = CFE_TBL_ERR_NO_ACCESS;
+
+    /* fail the register app */
+    Ut_CFE_TBL_SetReturnCode(UT_CFE_TBL_REGISTER_INDEX, expected, 1);
 
+    /* Execute the function being tested */
+    result = oSENS.InitApp();
+
+    /* Verify results */
+    UtAssert_True (result == expected,
+                   "InitConfigTbl, fail TBL Register");
 }
 
 
-void SENS_Config_Tbl_Test_AddTestCases(void)
+/**
+ * Test SENS InitConfigTbl(), fail ValidateConfigTbl
+ * Can not call private function, InitConfigTbl directly
+ * Currently there are no way to make this fail
+ */
+void Test_SENS_InitConfigTbl_Fail_ValidateConfigTbl(void)
+{
+    SENS  oSENS;
+    int32 result = CFE_SUCCESS;
+
+    /* Execute the function being tested */
+    result = oSENS.InitApp();
```

Comment:
Is there anything that can be validated?  Does this application even have a validate function?  If not, submit a ticket.  It should.

Reply:
It has a validate function, but there is no validation code in it. I submitted a ticket for this #272.

Reply:
And I changed the Fail_ValidateConfigTbl() to the ValidateConfigTbl_Nominal() and pushed.
Because I don't know the return value of the failure status. It looks like the return statuses are different dependent on the applications.